### ISSUE TITLE
Infusion duration set in $MAIN doesn't correctly adjust for reduced bioavailability

### ISSUE THREAD

**Issue by kylebmetrum** [metrumresearchgroup/mrgsolve#3]
$MAIN
    // amt = 1000, rate = -2;  actual rate should be 650/2  
    _F(1) = 0.65
    _D(1) = 2

This affects only infusions where duration is modeled (rate = -2) and bioavailability is modeled as well.
This will be fixed in the next release.

**Comment by kylebmetrum** [comment]
# 
  
  


```r
library(mrgsolve)
library(dplyr)
```

# D and F "in the model"


```r
code <- '
$PARAM D1 = 2, F1 = 0.5
$MAIN _D(1) = D1; _F(1) = F1;
$CMT DEPO CENT
$ODE dxdt_DEPO = -1.2*DEPO; dxdt_CENT = 1.2*DEPO-0.8*CENT;
'
```

```r
mod <- mread(code=code, "withD", tempdir())
out <- mod %>% ev(amt=1000, rate=-2) %>% mrgsim(delta=0.1, end=10,obsonly=TRUE) 
```

# D in the data set, F in the model


```r
code2 <- '
$PARAM  F1 = 0.5
$MAIN _F(1) = F1;
$CMT DEPO CENT
$ODE dxdt_DEPO = -1.2*DEPO; dxdt_CENT = 1.2*DEPO-0.8*CENT;
'
```

```r
mod2 <- mread(code=code2,"noD", tempdir())
out2 <- mod2 %>% ev(amt=1000, rate = amt/2) %>% mrgsim(delta=0.1,end=10,obsonly=TRUE) 
```

# Compare




This is right


```r
out  %>% as.tbl %>% slice(which.max(DEPO))
```

```
## Source: local data frame [1 x 4]
## 
##      ID  time     DEPO     CENT
##   (dbl) (dbl)    (dbl)    (dbl)
## 1     1     2 189.4338 179.9207
```

This isn't


```r
out2 %>% as.tbl %>% slice(which.max(DEPO))
```

```
## Source: local data frame [1 x 4]
## 
##      ID  time     DEPO    CENT
##   (dbl) (dbl)    (dbl)   (dbl)
## 1     1     1 291.1691 159.001
```

```r
identical(as.data.frame(out2),as.data.frame(out))
```

```
## [1] FALSE
```

# D and F in the data set


```r
code3 <- '
$PARAM  F1 = 0.5
$CMT DEPO CENT
$ODE dxdt_DEPO = -1.2*DEPO; dxdt_CENT = 1.2*DEPO-0.8*CENT;
'
```

```r
mod3 <- mread(code=code3,"noFnoD", tempdir())
out3 <- mod3 %>% 
  ev(amt =1000, amt=amt*param(mod3)$F1, rate = amt/2) %>% 
  mrgsim(delta=0.1,end=10,obsonly=TRUE)

out3 %>% as.tbl %>% slice(which.max(DEPO))
```

```
## Source: local data frame [1 x 4]
## 
##      ID  time     DEPO     CENT
##   (dbl) (dbl)    (dbl)    (dbl)
## 1     1     2 189.4338 179.9207
```

```r
identical(as.data.frame(out3),as.data.frame(out))
```

```
## [1] TRUE
```

**Comment by kylebmetrum** [comment]
Fixed in 0.5.11
